src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -450,6 +451,12 @@ def process_inputs(
         )
 
         generation_config = override_config(inputs.generation_kwargs, generation_config)
+        # If set to default, max_length is initially None. Update based on sequence
+        # length and prompt_sequence_length
+        if generation_config.max_length is None:
+            generation_config.max_length = (
+                self.sequence_length - self.prompt_sequence_length
```

Comment:
why are we using `prompt_sequence_length` here? `prompt_sequence_length` is the size of the chunk for prompt prefill. IMO this should be simply
`generation_config.max_length = self.sequence_length`.
Or am I missing something?
Could you maybe add some manual testing examples to the PR description, this could clear some of the potential confusion.

Reply:
few definitions here - @dbogunowicz I think you have a point that 1) max total tokens (ie `max_length`) should be our sequence length
2) what we're trying to compute is the max new tokens which should be `sequence_length - prompt_length` - since `prompt_sequence_length` is the chunk size (just for engine input, not necessarily true size) for prompt length we need to use `num_prompt_tokens`